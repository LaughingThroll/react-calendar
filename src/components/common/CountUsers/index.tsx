import React from 'react'
import styles from './users.module.scss'
import usersSvg from '../../../assets/images/icons/users.svg'

export interface CountUsersProps {
  countUsers: number
  src?: string
  classNames?: string[]
}

const { users, users__count, users__svg } = styles

const CountUsers: React.FC<CountUsersProps> = ({ src = usersSvg, countUsers, classNames = [] }) => {
  return (
    <div className={[users, ...classNames].join(' ')}>
      <img className={users__svg} src={src} alt="" />
      <span className={users__count}>{countUsers}</span>
    </div>
  )
}

export default React.memo(CountUsers)
