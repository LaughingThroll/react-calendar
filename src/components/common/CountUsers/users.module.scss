@import 'styles/utils/variables';

.users {
  display: flex;
  color: $users-count;

  &__count {
    margin-left: 3px;
  }
}
