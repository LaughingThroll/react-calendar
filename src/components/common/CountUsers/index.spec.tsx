import { shallow } from 'enzyme'
import CountUsers, { CountUsersProps } from '.'

const setUp = (props: CountUsersProps) => shallow(<CountUsers {...props} />)

describe('Users render', () => {
  it('default render', () => {
    const props = { countUsers: 10 }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('10')).toBe(true)
  })

  it('render with countUsers < 0', () => {
    const props = { countUsers: -10 }
    const tree = setUp(props)

    expect(tree.text().includes('-10')).toBe(true)
  })
})
