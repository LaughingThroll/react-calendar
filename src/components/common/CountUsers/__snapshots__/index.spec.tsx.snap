// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users render default render 1`] = `
<div
  className="users"
>
  <img
    alt=""
    className="users__svg"
    src="./../../../assets/users.svg"
  />
  <span
    className="users__count"
  >
    10
  </span>
</div>
`;
